deadlines/src/main/java/com/palantir/deadlines/Deadlines.java

```diff
@@ -75,7 +75,25 @@ private static Optional<RemainingDeadline> getRemainingDeadlineInternal() {
         // compute the remaining deadline relative to the current wall clock (may be negative)
         long elapsed = getClockNanoTime() - providedDeadline.wallClockNanos();
         long remaining = providedDeadline.valueNanos() - elapsed;
-        return Optional.of(new RemainingDeadline(remaining, providedDeadline.internal()));
+        return Optional.of(
+                new RemainingDeadline(remaining, providedDeadline.internal(), providedDeadline.disablePropagation()));
+    }
+
+    /**
+     * Disables propagation of deadline values any further for the current trace.
+     *
+     * Callers can use this to short-circuit deadline propagation from the current trace when they are sure that
+     * further operations should not be subject to deadline enforcement. Further calls to {@link #encodeToRequest}
+     * will result in a no-op assuming a deadline has previously be set for this trace (e.g. via a previous call to
+     * {@link #parseFromRequest}).
+     */
+    public static void disableFurtherDeadlinePropagation() {
```

Comment:
Should `Optional<Duration> getRemainingDeadline()` return an empty optional after this is called?

Reply:
i'm not sure. `disableFurtherDeadlinePropagation` doesn't mean that the deadline no longer exists in the current TraceLocal state, just that we won't add the header on future calls to `encodeToRequest`. maybe there is a valid use case for checking the deadline value even after we have disabled propagation?

Reply:
> Outside of metrics, I think this should be equivalent to `deadlineState.remove()`

i think that's perhaps fair, though it narrows some possibilities quite a bit.

Reply:
I've updated the implementation to return `Optional.empty()` in this case.